metrics/fidelity/utility.py

```diff
@@ -0,0 +1,174 @@
+import logging
+
+import numpy as np
+import torch
+
+
+logger = logging.getLogger(__name__)
+
+
+def compute_fidelity(prob_y_hat, prob_y_hat_alpha, fidelity_type="sufficiency", clip=True, dataset_level=False):
+    """
+    clip=False ERASER's definition of sufficiency and comprehensiveness Cite?? ask chenhao
+    Calculating the fidelity value given probabilities
+    :param prob_y_hat: P(y_hat/x)
+    :param prob_y_hat_alpha: P(y_hat/x,alpha) where text is reduced by alpha
+    :param fidelity_type: sufficiency or comprehensiveness
+    :param clip: true for clipping the fidelity values between 0 and 1
+    :param dataset_level: dataset level fidelity else instance level
+    :return: fidelity
+    """
+    if clip:
+        mean_difference = np.clip(np.array(prob_y_hat) - np.array(prob_y_hat_alpha), 0, 1)
+    else:
+        mean_difference = prob_y_hat - prob_y_hat_alpha
+    if dataset_level:
+        mean_difference = np.mean(mean_difference)
+    if fidelity_type == "sufficiency":
+        return 1 - mean_difference
+    else:
+        return mean_difference
+
+
+def itemwize_normalization(fidelity, null_difference, fidelity_type, clip, eps):
```

Comment:
change name to "normalize_item_fidelity". I generally prefer to use imperatives as function names.